src/ansys/aedt/core/visualization/post/common.py

```diff
@@ -0,0 +1,2417 @@
+    "Far Fields": ansys.aedt.core.visualization.report.field.FarField,
+    "Near Fields": ansys.aedt.core.visualization.report.field.NearField,
+    "Eye Diagram": ansys.aedt.core.visualization.report.eye.EyeDiagram,
+    "Statistical Eye": ansys.aedt.core.visualization.report.eye.AMIEyeDiagram,
+    "AMI Contour": ansys.aedt.core.visualization.report.eye.AMIConturEyeDiagram,
+    "Eigenmode Parameters": ansys.aedt.core.visualization.report.standard.Standard,
+    "Spectrum": ansys.aedt.core.visualization.report.standard.Spectral,
+    "EMIReceiver": ansys.aedt.core.visualization.report.emi.EMIReceiver,
+}
+
+
+class PostProcessorCommon(object):
+    """Manages the main AEDT postprocessing functions.
+
+    This class is inherited in the caller application and is accessible through the post variable( eg. ``hfss.post`` or
+    ``q3d.post``).
+
+    .. note::
+       Some functionalities are available only when AEDT is running in
+       the graphical mode.
+
+    Parameters
+    ----------
+    app : :class:`ansys.aedt.core.application.analysis_3d.FieldAnalysis3D`
+        Inherited parent object. The parent object must provide the members
+        ``_modeler``, ``_desktop``, ``_odesign``, and ``logger``.
+
+    Examples
+    --------
+    >>> from ansys.aedt.core import Q3d
+    >>> q3d = Q3d()
+    >>> q.post.get_solution_data(domain="Original")
+    """
+
+    def __init__(self, app):
+        self._app = app
+        self.oeditor = None
+        if self.modeler:
+            self.oeditor = self.modeler.oeditor
+        self._scratch = self._app.working_directory
+        self.plots = self._get_plot_inputs()
+        self.reports_by_category = Reports(self, self._app.design_type)
+
+    @property
+    def available_report_types(self):
+        """Report types.
+
+        References
+        ----------
+
+        >>> oModule.GetAvailableReportTypes
+        """
+        return list(self.oreportsetup.GetAvailableReportTypes())
+
+    @property
+    def update_report_dynamically(self):
+        """Get/Set the boolean to automatically update reports on edits.
+
+        Returns
+        -------
+        bool
+        """
+        return (
+            True
+            if self._app.odesktop.GetRegistryInt(
+                "Desktop/Settings/ProjectOptions/{}/UpdateReportsDynamicallyOnEdits".format(self._app.design_type)
+            )
+            == 1
+            else False
+        )
+
+    @update_report_dynamically.setter
+    def update_report_dynamically(self, value):
+        if value:
+            self._app.odesktop.SetRegistryInt(
+                "Desktop/Settings/ProjectOptions/{}/UpdateReportsDynamicallyOnEdits".format(self._app.design_type), 1
+            )
+        else:  # pragma: no cover
+            self._app.odesktop.SetRegistryInt(
+                "Desktop/Settings/ProjectOptions/{}/UpdateReportsDynamicallyOnEdits".format(self._app.design_type), 0
+            )
```

Comment:
```suggestion
                f"Desktop/Settings/ProjectOptions/{self._app.design_type}/UpdateReportsDynamicallyOnEdits", 1
            )
        else:  # pragma: no cover
            self._app.odesktop.SetRegistryInt(
                f"Desktop/Settings/ProjectOptions/{self._app.design_type}/UpdateReportsDynamicallyOnEdits", 0
            )
```
We can use f-strings now ! :)